src/Services/FHIR/FhirAllergyIntoleranceService.php

```diff
@@ -0,0 +1,178 @@
+        $result['code'] = $code['dx_code'];
+        return $result;        
+    }
+
+    public function createAllergyIntoleranceResource($id = '', $data = '', $encode = true)
+    {     
+        $type = new FHIRAllergyIntoleranceType();
+        $type->setValue($data['type']);
+
+        $onSetDateTime = new FHIRDateTime();
+        $onSetDateTime->setValue($data['begdate']);
+
+        $recordedDate = new FHIRDateTime();
+        $recordedDate->setValue($data['recorded_date']);
+        
+        $clinicalStatus = '';
+        if ($data['outcome'] == '1' && isset($data['enddate']))
+            $clinicalStatus = "resolved";
+        else if (isset($data['enddate'])) 
+            $clinicalStatus = "active";
+        else
+            $clinicalStatus = "inactive";
+        $clinicalStatusCoding = new FHIRCoding();
+        $clinicalStatusCoding->setSystem("http://terminology.hl7.org/CodeSystem/allergyintolerance-clinical");
+        $clinicalStatusCoding->setCode($clinicalStatus);
+        $clinicalStatusCoding->setDisplay($clinicalStatus);
+
+        $category = new FHIRAllergyIntoleranceCategory();
+        $category->setValue($data['subtype']);
+
+        $criticality = new FHIRAllergyIntoleranceCriticality();
+        $criticality->setValue($data['severity_al']);
+
+        $patient = new FHIRReference();
+        $patient->setReference('Patient/'.$data['pid']);
+
+        $recorder = new FHIRReference();
+        $recorder->setReference('Practitioner/'.$data['referredby']);
+
+        $asserter = new FHIRReference();
+        $asserter->setReference('Patient/'.$data['pid']);
+
+        $note = new FHIRAnnotation();
+        $note->setText($data['extrainfo']);
+
+        $lastOccurrence = new FHIRDateTime();
+        $lastOccurrence->setValue($data['returndate']);
+
+        $manifestation = new FHIRCoding();
+        $manifestation->setSystem("http://hl7.org/fhir/sid/icd-10-cm");
+        $manifestation->setCode($data['code']);
+        $manifestation->setDisplay($data['reaction']);
+
+        $description = $data['comments'];
+
+        $reaction = new FHIRAllergyIntoleranceReaction();
+        $reaction->addManifestation($manifestation->jsonSerialize());
+        $reaction->setDescription($description);
+        $reaction->setOnset($data['begdate']);
+        $reaction->setSeverity($data['severity_al']);
+        
+        $resource = new FHIRAllergyIntolerance();
+        $resource->setClinicalStatus($clinicalStatusCoding);
+        $resource->setType($type);
+        $resource->addCategory($category);
+        $resource->setOnsetDateTime($onSetDateTime);
+        $resource->setRecordedDate($recordedDate);
+        $resource->setCriticality($criticality);
+        $resource->setPatient($patient);
+        $resource->setRecorder($recorder);
+        $resource->setAsserter($asserter);
+        $resource->setLastOccurrence($lastOccurrence);
+        $resource->addNote($note);
+        $resource->addReaction($reaction);
```

Comment:
Thank you. I know this method is more tedious but it reads so much better than building the json outside the class.